ext/opentelemetry-ext-cloud-trace/src/opentelemetry/ext/cloud_trace/__init__.py

```diff
@@ -0,0 +1,254 @@
+            except Exception as ex:
+                logger.warning("Error %s when creating span %s", ex, span)
+
+        try:
+            self.client.batch_write_spans(
+                "projects/{}".format(self.project_id), cloud_trace_spans,
+            )
+        # pylint: disable=broad-except
+        except Exception as ex:
+            logger.warning("Error while writing to Cloud Trace: %s", ex)
+            return SpanExportResult.FAILURE
+
+        return SpanExportResult.SUCCESS
+
+    def translate_to_cloud_trace(
+        self, spans: Sequence[Span]
+    ) -> List[Dict[str, Any]]:
+        """Translate the spans to Cloud Trace format.
+
+        Args:
+            spans: Tuple of spans to convert
+        """
+
+        cloud_trace_spans = []
+
+        for span in spans:
+            ctx = span.get_context()
+            trace_id = _get_hexidecimal_trace_id(ctx.trace_id)
+            span_id = _get_hexidecimal_span_id(ctx.span_id)
+            span_name = "projects/{}/traces/{}/spans/{}".format(
+                self.project_id, trace_id, span_id
+            )
+
+            parent_id = None
+            if isinstance(span.parent, trace_api.Span):
+                parent_id = _get_hexidecimal_span_id(
+                    span.parent.get_context().span_id
+                )
+            elif isinstance(span.parent, trace_api.SpanContext):
+                parent_id = _get_hexidecimal_span_id(span.parent.span_id)
+
+            start_time = get_time_from_ns(span.start_time)
+            end_time = get_time_from_ns(span.end_time)
+
+            attributes = _extract_attributes(span.attributes)
+
+            cloud_trace_spans.append(
+                {
+                    "name": span_name,
+                    "span_id": span_id,
+                    "display_name": get_truncatable_str_object(span.name),
+                    "start_time": start_time,
+                    "end_time": end_time,
+                    "parent_span_id": parent_id,
+                    "attributes": attributes,
+                    "links": _extract_links(span.links),
+                    "status": _extract_status(span.status),
+                    "time_events": _extract_events(span.events),
+                }
+            )
+
+        return cloud_trace_spans
+
+    def shutdown(self):
+        pass
+
+
+def _get_hexidecimal_trace_id(trace_id: int) -> str:
+    return "{:032x}".format(trace_id)
+
+
+def _get_hexidecimal_span_id(span_id: int) -> str:
+    return "{:016x}".format(span_id)
+
+
+def get_time_from_ns(nanoseconds):
+    """Given epoch nanoseconds, split into epoch milliseconds and remaining nanoseconds"""
+    if not nanoseconds:
+        return None
+    return {"seconds": int(nanoseconds / 1e9), "nanos": int(nanoseconds % 1e9)}
+
+
+def get_truncatable_str_object(str_to_convert, max_length=MAX_LENGTH):
+    """Truncate the string if it exceeds the length limit and record the truncated bytes
+        count.
+    """
```

Comment:
```suggestion
    """Truncate the string if it exceeds the length limit and record the
    truncated bytes count.
    """
```

Some docstring formatting nits: we generally wrap docstrings and comments at 80 chars and only indent wrapped lines when the indentation is semantically meaningful (e.g. documenting args), and otherwise follow [PEP 257](https://www.python.org/dev/peps/pep-0257/).

Ideally the linter/formatter would handle this for us and we'd never have to make this kind of edit in a review, but [black](https://github.com/psf/black) only gets us so far.